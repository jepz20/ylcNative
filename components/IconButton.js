// @flow

import * as React from 'react'
import { Button } from './Button'
import { Entypo } from '@expo/vector-icons'

type Props = {
  name: string,
  size: number,
  color: string
}

const IconButton = ({ name, size, color, ...props }: Props) => {
  return (
    <Button {...props}>
      <Entypo name={name} size={size} color={color} />
    </Button>
  )
}

IconButton.defaultProps = {
  size: 40,
  color: '#FFF'
}

export { IconButton }
