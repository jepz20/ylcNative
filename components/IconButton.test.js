import 'react-native'
import React from 'react'
import { IconButton } from './IconButton'

import renderer from 'react-test-renderer'

describe('IconButton', () => {
  test('Renders a button with a icon', () => {
    const iconButton = renderer.create(<IconButton name='plus' />).toJSON()
    expect(iconButton).toMatchSnapshot()
  })

  test('Renders a button with default size', () => {
    const iconButton = renderer
      .create(<IconButton name='plus' size='default' />)
      .toJSON()
    expect(iconButton).toMatchSnapshot()
  })

  test('doesnt renders a children', () => {
    const iconButton = renderer
      .create(<IconButton>Shouldn't render</IconButton>)
      .toJSON()
    expect(iconButton).toMatchSnapshot()
  })
})
