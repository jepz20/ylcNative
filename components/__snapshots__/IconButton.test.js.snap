// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton Renders a button with a icon 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  />
</View>
`;

exports[`IconButton Renders a button with default size 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#fff",
          "fontSize": 12,
        },
        null,
        Object {
          "fontFamily": "entypo",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    ïˆš
  </Text>
</View>
`;

exports[`IconButton doesnt renders a children 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#fff",
          "fontSize": 40,
        },
        null,
        Object {
          "fontFamily": "entypo",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
</View>
`;
